autogpt_platform/backend/backend/data/graph.py

```diff
@@ -102,88 +95,93 @@ def from_agent_graph_execution(execution: AgentGraphExecution):
         )
 
 
-class GraphMeta(BaseDbModel):
+class Graph(BaseDbModel):
     version: int = 1
     is_active: bool = True
     is_template: bool = False
     name: str
     description: str
-    executions: list[ExecutionMeta] | None = None
+    executions: list[GraphExecution] = []
+    nodes: list[Node] = []
+    links: list[Link] = []
 
     @staticmethod
-    def from_db(graph: AgentGraph):
-        if graph.AgentGraphExecution:
-            executions = [
-                ExecutionMeta.from_agent_graph_execution(execution)
-                for execution in graph.AgentGraphExecution
-            ]
-        else:
-            executions = None
-
-        return GraphMeta(
-            id=graph.id,
-            version=graph.version,
-            is_active=graph.isActive,
-            is_template=graph.isTemplate,
-            name=graph.name or "",
-            description=graph.description or "",
-            executions=executions,
-        )
+    def _generate_schema(
+        type_class: Type[AgentInputBlock.Input] | Type[AgentOutputBlock.Input],
+        data: list[dict],
+    ) -> dict[str, Any]:
```

Comment:
nit: I wonder if typing could improve here

Reply:
This is unfortunately coming from pydantic `model_json_schema`:
<img width="642" alt="image" src="https://github.com/user-attachments/assets/d41e77a3-58c2-4d94-9177-57b37d8d61cf">

They don't have better typing for this since it's a dynamic JSONschema structure.